source/common/http/url_utility.cc

```diff
@@ -0,0 +1,95 @@
+#include "common/http/url_utility.h"
+
+#include <sys/types.h>
+
+#include <iostream>
+#include <string>
+
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+#include "common/common/utility.h"
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Http {
+namespace Utility {
+
+bool Url::initialize(absl::string_view absolute_url, bool is_connect) {
```

Comment:
@htuch: At some point, can we get rid of https://github.com/envoyproxy/envoy/tree/master/source/common/chromium_url by doing the normalization in this class via GURL?
@dio: I think so yeah.


Ref: https://github.com/envoyproxy/envoy/pull/10599#discussion_r403337765